### PR TITLE
Added Capability to find nearest walkable path to a point.

### PR DESCRIPTION
**Pull request by arekmano**
- TiledMap can now return the nearest walkable path
- Uses breadth first search to do so
- Added toString for debugging purposes

### CODE REVIEW

File: app/src/main/java/com/concordia/mcga/utilities/pathfinding/TiledMap.java [hunk 1]
```diff
@@ -38,12 +49,66 @@ public void setTile(int x, int y, PathFinderTile tile) {
...
     /**
      * Sets the given X,Y coordinate as walkable
+     *
      * @param indoorMapTile -
      */
     public void makeWalkable(IndoorMapTile indoorMapTile) {
         allTiles[indoorMapTile.getCoordinateX()][indoorMapTile.getCoordinateY()] = new PathFinderTile(indoorMapTile);
     }
 
+    public boolean isWalkable(IndoorMapTile indoorMapTile) {
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/utilities/pathfinding/TiledMap.java]: javadoc

File: app/src/main/java/com/concordia/mcga/utilities/pathfinding/TiledMap.java [hunk 2]
```diff
@@ -38,12 +49,66 @@ public void setTile(int x, int y, PathFinderTile tile) {
...
+        return getTile(indoorMapTile.getCoordinateX(), indoorMapTile.getCoordinateY()) != null;
+    }
+
+    public PathFinderTile closestWalkable(IndoorMapTile indoorMapTile) throws MCGAPathFindingException {
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/utilities/pathfinding/TiledMap.java]: javadoc

**Comment by Taimoorrana1** [comment]
nice test coverage

File: app/src/main/java/com/concordia/mcga/utilities/pathfinding/TiledMap.java [hunk 3]
```diff
@@ -38,12 +49,66 @@ public void setTile(int x, int y, PathFinderTile tile) {
...
+     * @param coordinateX X coordinate to use for the breadth first search algorithm
+     * @param coordinateY Y coordinate to use for the breadth first search algorithm
+     * @return The closest PathFinderTile to the current given coordinates. Looks at tiles
+     *         that are directly above, below, to the left and to the right. Does not look at
+     *         tiles that are diagonal to the given coordinates.
+     * @throws MCGAPathFindingException Thrown when no valid coordinate can be found.
+     */
+    PathFinderTile findBreadthFirstSearch(int coordinateX, int coordinateY) throws MCGAPathFindingException {
+        int delta = 1;
+        int limit = Math.max(allTiles.length, allTiles[0].length);
```
Comment (adriannadiaz) [inline app/src/main/java/com/concordia/mcga/utilities/pathfinding/TiledMap.java]: Either make this variable final or put the expression directly inside the while statement

**Pull request merged**